File: mbw/src/main/java/com/mycelium/wallet/activity/settings/NotificationsFragment.java
package com.mycelium.wallet.activity.settings;

import android.os.Bundle;
import android.support.v7.app.ActionBar;
import android.support.v7.preference.PreferenceFragmentCompat;
import android.view.MenuItem;

import com.mycelium.wallet.R;

public class NotificationsFragment extends PreferenceFragmentCompat {
    @Override
    public void onCreatePreferences(Bundle savedInstanceState, String rootKey) {
        addPreferencesFromResource(R.xml.preferences_notifications);

        setHasOptionsMenu(true);
        ActionBar actionBar = ((SettingsActivity) getActivity()).getSupportActionBar();
        actionBar.setTitle(R.string.notifications);
        actionBar.setHomeAsUpIndicator(R.drawable.ic_back_arrow);
        actionBar.setDisplayShowHomeEnabled(false);
        actionBar.setDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            getFragmentManager().popBackStack();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }
}

File: mbw/src/main/java/com/mycelium/wallet/activity/settings/NotificationsFragment.kt
package com.mycelium.wallet.activity.settings

import android.os.Bundle
import android.view.MenuItem
import androidx.preference.CheckBoxPreference
import androidx.preference.PreferenceFragmentCompat
import com.mycelium.wallet.R

class NotificationsFragment : PreferenceFragmentCompat() {
    private var newsAllPreference: CheckBoxPreference? = null

    override fun onCreatePreferences(savedInstanceState: Bundle, rootKey: String) {
        addPreferencesFromResource(R.xml.preferences_notifications)

        setHasOptionsMenu(true)
        (activity as SettingsActivity).supportActionBar?.run {
            setTitle(R.string.notifications)
            setHomeAsUpIndicator(R.drawable.ic_back_arrow)
            setDisplayShowHomeEnabled(false)
            setDisplayHomeAsUpEnabled(true)
        }

        newsAllPreference = findPreference("news_all_notification")
    }

    override fun onBindPreferences() {
        newsAllPreference?.isChecked = SettingsPreference.mediaFLowNotificationEnabled
        newsAllPreference?.setOnPreferenceClickListener { preference ->
            val p = preference as CheckBoxPreference
            SettingsPreference.mediaFLowNotificationEnabled = p.isChecked
            true
        }
    }

    override fun onOptionsItemSelected(item: MenuItem): Boolean {
        if (item.itemId == android.R.id.home) {
            fragmentManager?.popBackStack()
            return true
        }
        return super.onOptionsItemSelected(item)
    }
}
